README.pt-br.md

```diff
@@ -1296,38 +1306,132 @@ get '/:value' do
+de como você pode criar um segredo de 64 bytes e usá-lo:
+
+**Gerando o Segredo da Sessão**
+
+```text
+$ ruby -e "require 'securerandom'; puts SecureRandom.hex(64)"
+99ae8af...snip...ec0f262ac
 ```
 
-Para melhorar a segurança, os dados da sessão guardados no cookie é assinado com uma chave secreta da sessão. Uma chave aleatória é gerada para você pelo Sinatra. Contudo, já que a chave mudará cada vez que você inicia sua aplicação, você talvez queira defini-la você mesmo para que todas as instâncias da aplicação compartilhe-a:
+**Gerando o Segredo da Sessão (Bônus)**
+
+Use o [sysrandomgem](https://github.com/cryptosphere/sysrandom#readme) para usar
+as facilidades do sistema RNG de gerar valores aleatórios em vez do userspace
+`OpenSSL` que é o padrão atual do MRI Ruby:
+
+```text
+$ gem install sysrandom
+Building native extensions.  This could take a while...
+Successfully installed sysrandom-1.x
+1 gem installed
+
+$ ruby -e "require 'sysrandom/securerandom'; puts SecureRandom.hex(64)"
+99ae8af...snip...ec0f262ac
+```
+
+**Variável de Ambiente do Segredo da Sessão**
+
+Crie uma variável de ambiente `SESSION_SECRET` para o Sinatra com o valor que
+foi gerado por você. Faça com que esse valor fique persistente através das
+reinicializações do seu host. O método para fazer isso vai variar de sistema
+para sistema, este é apenas para propósitos ilustrativos:
+
+```bash
+# echo "export SESSION_SECRET=99ae8af...snip...ec0f262ac" >> ~/.bashrc
+```
+
+**Session Secret App Config**
+
+Configure sua aplicação para torná-la a prova de falhas caso o segredo aleatório
+seguro não esteja presente na variável de ambiente `SESSION_SECRET`.
+
+Para pontos de bônus use o
+[sysrandomgem](https://github.com/cryptosphere/sysrandom#readme) como é feito
+aqui:
 
 ```ruby
-set :session_secret, 'super secret'
+require 'securerandom'
+# -ou- require 'sysrandom/securerandom'
+set :session_secret, ENV.fetch('SESSION_SECRET') { SecureRandom.hex(64) }
 ```
 
-Se você quiser fazer outras configurações, você também pode guardar um hash com as opções nas configurações da `session`:
+#### Configuração da Sessão
+
+Se você quiser configurar ainda mais, você também pode armazenar em um hash com
+opções dentro das configurações de `sessions`:
 
 ```ruby
 set :sessions, :domain => 'foo.com'
 ```
 
-Para compartilhar sua sessão entre outros aplicativos em um subdomínio de foo.com, utilize o prefixo *.*:
+Para compartilhar sua sessão entre outros aplicativos em um subdomínio de
+foo.com, utilize o prefixo *.*:
 
 ```ruby
 set :sessions, :domain => '.foo.com'
 ```
 
+#### Escolhendo seu Próprio Middleware de Sessão
+
+Note que `enable :sessions` na verdade guarda todos os dados em um cookie. Isso
+pode não ser sempre o que você deseja (salvar muitos dados vai aumentar seu
+tráfego, por exemplo). Você pode usar qualquer middleware de sessão do Rack para
+fazer isso, um dos seguinte métodos pode ser usados:
```

Comment:
We don't need the plural here, as the number has to agree with the "um" from before:
```suggestion
fazer isso, um dos seguintes métodos pode ser usado:
```